### PR TITLE
GDScript: Add error messages to some compilation errors

### PR DESCRIPTION
**Pull request by rune-scape**
added some messages to avoid empty compiler errors

### CODE REVIEW

**Comment by Calinou** [comment]
If these bugs are not *meant* to ever be encountered by users, I suggest suffixing them with "(please report)".

**Comment by rune-scape** [comment]
@Calinou done :+1:

File: modules/gdscript/gdscript_compiler.cpp [hunk 1]
```diff
@@ -3259,7 +3278,12 @@ Error GDScriptCompiler::compile(const GDScriptParser *p_parser, GDScript *p_scri
 		GDScriptCache::add_static_script(p_script);
 	}
 
-	return GDScriptCache::finish_compiling(main_script->path);
+	err = GDScriptCache::finish_compiling(main_script->path);
+	if (err) {
+		_set_error(R"(Failed to compile depended scripts.)", nullptr);
+		return err;
```
Comment (AThousandShips) [inline modules/gdscript/gdscript_compiler.cpp:3284]:
```suggestion
```
Redundant, or make the below return `OK`

**Review by Calinou — APPROVED** [review]

**Review by adamscott — APPROVED** [review]

File: modules/gdscript/gdscript_compiler.cpp [hunk 2]
```diff
@@ -726,6 +727,7 @@ GDScriptCodeGenerator::Address GDScriptCompiler::_parse_expression(CodeGen &code
 							return GDScriptCodeGenerator::Address();
 						}
 					} else {
+						_set_error("Compiler bug (please report): incorrect callee type in call node!", call->callee);
```
[the remark this answers is not in the document]
Comment (Calinou) [reply modules/gdscript/gdscript_compiler.cpp:730]: Yes, it makes more sense to use a period.

File: modules/gdscript/gdscript_compiler.cpp [hunk 3]
```diff
@@ -2721,7 +2739,8 @@ Error GDScriptCompiler::_prepare_compilation(GDScript *p_script, const GDScriptP
 		case GDScriptDataType::GDSCRIPT: {
 			Ref<GDScript> base = Ref<GDScript>(base_type.script_type);
 			if (base.is_null()) {
-				return ERR_COMPILATION_FAILED;
+				_set_error("Compiler bug (please report): base script type is null!", nullptr);
+				return ERR_BUG;
```
[the remark this answers is not in the document]
Comment (Calinou) [reply modules/gdscript/gdscript_compiler.cpp:2743]: Same as above, I'd use a `.`.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**